### PR TITLE
Configuring heroku database when using postgres

### LINKED ISSUE #2876: Code Broken : HerokuDatabaseConfiguration
Using jhipster 2.27.0 apoint to master, the code generated for 'yo jhipster:heroku' create the class HerokuDatabaseConfiguration using jHipsterProperties.getDatasource() in the code, but this method was removed.

``` JAVA
Error: Command failed: /bin/sh -c ./gradlew -Pprod bootRepackage -x test && heroku deploy:jar --jar build/libs/*.war
npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.7
/mnt/dados/projetos/comment/codigo/api/src/main/java/me/comment/api/config/HerokuDatabaseConfiguration.java:30: error: cannot find symbol
                config.addDataSourceProperty("cachePrepStmts", jHipsterProperties.getDatasource().isCachePrepStmts());
                                                                                 ^
  symbol:   method getDatasource()
  location: variable jHipsterProperties of type JHipsterProperties
/mnt/dados/projetos/comment/codigo/api/src/main/java/me/comment/api/config/HerokuDatabaseConfiguration.java:31: error: cannot find symbol
                config.addDataSourceProperty("prepStmtCacheSize", jHipsterProperties.getDatasource().getPrepStmtCacheSize());
                                                                                    ^
  symbol:   method getDatasource()
  location: variable jHipsterProperties of type JHipsterProperties
/mnt/dados/projetos/comment/codigo/api/src/main/java/me/comment/api/config/HerokuDatabaseConfiguration.java:32: error: cannot find symbol
                config.addDataSourceProperty("prepStmtCacheSqlLimit", jHipsterProperties.getDatasource().getPrepStmtCacheSqlLimit());
                                                                                        ^
  symbol:   method getDatasource()
  location: variable jHipsterProperties of type JHipsterProperties
3 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

```

### PR DESCRIPTION
**Pull request by phouverneyuff**
Fix #2876 

If not set spring.datasource.driver-class-name, in heroku give the error:
```
ClassCastException: Cannot cast org.postgresql.Driver to javax.sql.DataSource
2016-02-14T19:36:17.056837+00:00 app[web.1]: 	at com.zaxxer.hikari.pool.PoolElf.initializeDataSource(PoolElf.java:153) ~[HikariCP-2.4.1.jar!/:na]
2016-02-14T19:36:17.056837+00:00 app[web.1]: 	at com.zaxxer.hikari.util.UtilityElf.createInstance(UtilityElf.java:79) ~[HikariCP-2.4.1.jar!/:na]
2016-02-14T19:36:17.056835+00:00 app[web.1]: 	... 38 common frames omitted
2016-02-14T19:36:17.056836+00:00 app[web.1]: Caused by: java.lang.RuntimeException: java.lang.ClassCastException: Cannot cast org.postgresql.Driver to javax.sql.DataSource
2016-02-14T19:36:17.056838+00:00 app[web.1]: 	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:113) ~[HikariCP-2.4.1.jar!/:na]
2016-02-14T19:36:17.056839+00:00 app[web.1]: 	at com.zaxxer.hikari.HikariDataSource.<init>(HikariDataSource.java:73) ~[HikariCP-2.4.1.jar!/:na]

```

### CODE REVIEW

File: generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java [hunk 1]
```diff
@@ -23,15 +23,16 @@ public DataSource dataSource(DataSourceProperties dataSourceProperties, JHipster
 
         String herokuUrl = System.getenv("JDBC_DATABASE_URL");
         if (herokuUrl != null) {
-	    HikariConfig config = new HikariConfig();
+	        HikariConfig config = new HikariConfig();
```
Comment (deepu105) [inline generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java]: Spacing?

File: generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java [hunk 2]
```diff
@@ -23,15 +23,16 @@ public DataSource dataSource(DataSourceProperties dataSourceProperties, JHipster
...
-	    if ("com.mysql.jdbc.jdbc2.optional.MysqlDataSource".equals(dataSourceProperties.getDriverClassName())) {
+	        <%_ if (prodDatabaseType == 'mysql') { _%>
+            //MySQL optimizations, see https://github.com/brettwooldridge/HikariCP/wiki/MySQL-Configuration
+    	    if ("com.mysql.jdbc.jdbc2.optional.MysqlDataSource".equals(dataSourceProperties.getDriverClassName())) {
                 config.addDataSourceProperty("cachePrepStmts", jHipsterProperties.getDatasource().isCachePrepStmts());
                 config.addDataSourceProperty("prepStmtCacheSize", jHipsterProperties.getDatasource().getPrepStmtCacheSize());
                 config.addDataSourceProperty("prepStmtCacheSqlLimit", jHipsterProperties.getDatasource().getPrepStmtCacheSqlLimit());
             }
-
+            <%_ } _%>
```
Comment (deepu105) [inline generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java]: Unwanted spaces

File: generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java [hunk 3]
```diff
@@ -23,15 +23,16 @@ public DataSource dataSource(DataSourceProperties dataSourceProperties, JHipster
 
         String herokuUrl = System.getenv("JDBC_DATABASE_URL");
         if (herokuUrl != null) {
-	    HikariConfig config = new HikariConfig();
+	        HikariConfig config = new HikariConfig();
 
-	    //MySQL optimizations, see https://github.com/brettwooldridge/HikariCP/wiki/MySQL-Configuration
-	    if ("com.mysql.jdbc.jdbc2.optional.MysqlDataSource".equals(dataSourceProperties.getDriverClassName())) {
+	        <%_ if (prodDatabaseType == 'mysql') { _%>
```
Comment (deepu105) [inline generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java]: Spacing

File: generators/server/templates/src/main/resources/config/_application-prod.yml [hunk 4]
```diff
@@ -48,6 +48,9 @@ spring:
         username: <%= baseName %>
         <%_ } _%>
         password:
+        <%_ if (prodDatabaseType == 'postgresql') { _%>
```
Comment (deepu105) [inline generators/server/templates/src/main/resources/config/_application-prod.yml]: Ping @cbornet

File: generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java (hunk 3 above)
Comment (cbornet) [inline generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java]: The mysql properties are now sent from the property file. So they must be removed from here and @ConfigurationProperties must be used

File: generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java (hunk 1 above)
Comment (cbornet) [inline generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java]: The Spring Boot DatabaseBuilder shall be used

File: generators/server/templates/src/main/resources/config/_application-prod.yml (hunk 4 above)
Comment (cbornet) [inline generators/server/templates/src/main/resources/config/_application-prod.yml]: This was removed on purpose. Please use the DatabaseBuilder. See _DatabaseConfiguration.java from the main generator.

File: generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java (hunk 1 above)
Comment (phouverneyuff) [inline generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java]: @deepu105 , the indentation was wrong.
@cbornet , I'll try use.

**Comment by cbornet** [comment]
Closing in favor of #2906

**Pull request closed without merging**